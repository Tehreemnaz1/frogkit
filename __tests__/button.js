import { mount } from 'vue-test-utils'
import Vue from 'vue'
import Button from '@/components/button'

describe('Button component', () => {
  it('should render component and match snapshot', () => {
    const wrapper = mount(Button)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render primary button', () => {
    const wrapper = mount(Button, { propsData: { color: 'primary' } })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render secondary button', () => {
    const wrapper = mount(Button, { propsData: { color: 'secondary' } })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render small button', () => {
    const wrapper = mount(Button, { propsData: { size: 'small' } })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render big button', () => {
    const wrapper = mount(Button, { propsData: { size: 'big' } })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render outline button', () => {
    const wrapper = mount(Button, { propsData: { outline: true } })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render block button', () => {
    const wrapper = mount(Button, { propsData: { block: true } })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render disabled button', () => {
    const wrapper = mount(Button, { propsData: { disabled: true } })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Button should emit click event when clicked', () => {
    const stub = jest.fn()
    const wrapper = mount(Button)
    wrapper.vm.$on('click', stub)
    wrapper.trigger('click')
    expect(stub).toBeCalled()
  })

})
