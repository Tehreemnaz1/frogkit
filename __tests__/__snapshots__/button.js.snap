// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render big button 1`] = `<button rel="noopener noreferrer" class="fk-btn fk-btn--big"><span>click me</span></button>`;

exports[`Button component should render block button 1`] = `<button rel="noopener noreferrer" class="fk-btn fk-btn--block"><span>click me</span></button>`;

exports[`Button component should render component and match snapshot 1`] = `<button rel="noopener noreferrer" class="fk-btn"><span>click me</span></button>`;

exports[`Button component should render disabled button 1`] = `<button rel="noopener noreferrer" class="fk-btn"><span>click me</span></button>`;

exports[`Button component should render outline button 1`] = `<button rel="noopener noreferrer" class="fk-btn fk-btn--outline"><span>click me</span></button>`;

exports[`Button component should render primary button 1`] = `<button rel="noopener noreferrer" class="fk-btn fk-btn--primary"><span>click me</span></button>`;

exports[`Button component should render secondary button 1`] = `<button rel="noopener noreferrer" class="fk-btn fk-btn--secondary"><span>click me</span></button>`;

exports[`Button component should render small button 1`] = `<button rel="noopener noreferrer" class="fk-btn fk-btn--small"><span>click me</span></button>`;
